### PR TITLE
Modified write_graphite to update it to new format for graphite nodes.

### PR DESCRIPTION
**Pull request by SPoage**
Also corrected an error in the pillar example file and added the write_graphite plugin to the list of states available in the README.

### CODE REVIEW

File: collectd/files/write_graphite.conf [hunk 1]
```diff
@@ -10,16 +10,18 @@
...
-   Postfix "{{ collectd_settings.plugins.write_graphite.postfix }}"
-   Protocol "{{ collectd_settings.plugins.write_graphite.protocol }}"
-   EscapeCharacter "{{ collectd_settings.plugins.write_graphite.escapecharacter }}"
-   LogSendErrors {{ collectd_settings.plugins.write_graphite.logsenderrors|lower }}
-   SeparateInstances {{ collectd_settings.plugins.write_graphite.separateinstances|lower }}
-   StoreRates {{ collectd_settings.plugins.write_graphite.storerates|lower }}
-   AlwaysAppendDS {{ collectd_settings.plugins.write_graphite.alwaysappendds|lower }}
- </Carbon>
+{%- for node_name, graphite_node in collectd_settings.plugins.write_graphite.nodes.iteritems() %}
+ <Node "{{ node_name }}">
```
Comment (iggy) [inline collectd/files/write_graphite.conf]: You need to version key this or just stick with Carbon for now. Some versions (notably the version of collectd in debian stable) don't understand <Node> (vs <Carbon>). FWIW, the code in collectd master still accepts Carbon.

**Comment by iggy** [comment]
In general, you want to split unrelated changes out into different PRs. I'm not sure that should hold this up, but something to keep in mind in the future.

File: collectd/files/write_graphite.conf (hunk 1 above)
Comment (SPoage) [inline collectd/files/write_graphite.conf]: Thanks, was not aware of that. I suppose I shouldn't be surprised - Debian tends to keep very old/stable versions around. Will attempt to update later to include a version check. Do you know if the old <Carbon> format still accepts multiple nodes? Or will it die if it gets more than one?

Comment (iggy) [inline collectd/files/write_graphite.conf]: I don't have 2 graphite servers to test, but collectd at least starts fine with 2 <Carbon> sections in there. Code wise, it look like it should work the same.

Comment (SPoage) [inline collectd/files/write_graphite.conf]: This hurts my pride to admit, but I can't seem to figure out how to go about this. I could have sworn I saw something similar in the Salt docs, but my google-fu is failing me. Any chance you have a project that implements such a pattern that I could reference? Or the a place in the docs that might be of relevance.

Comment (iggy) [inline collectd/files/write_graphite.conf]: The only thing I'm coming up with is a setting that goes in map.jinja. Default it to `new_graph_config: False` and then set it to True using another filter_by map that filters on lsb_codename (to differentiate the versions).

There are probably better ways.

File: README.rst [hunk 2]
```diff
@@ -80,6 +80,11 @@ Enables and configures the postgresql plugin. Needs refinement.
 
 Enables and configures the syslog plugin.
 
+``collectd.write_graphite``
+------------
```
Comment (nmadhok) [inline README.rst]: This needs to cover the whole of previous line's text. Something like the following:
```
``collectd.write_graphite``
---------------------------------
```

File: README.rst [hunk 3]
```diff
@@ -80,6 +80,11 @@ Enables and configures the postgresql plugin. Needs refinement.
...
+
+Enables and configures the write_graphite plugin.
+
 Usage
 ================
```
Comment (nmadhok) [inline README.rst]: Please remove extra ``=``
```
Usage
 ====
```

File: pillar.example [hunk 4]
```diff
@@ -1,8 +1,8 @@
 collectd:
   FQDNLookup: true
   TypesDB: ['/usr/share/collectd/types.db']
+  default_plugins: [battery, cpu, entropy, load, memory, swap, users]
```
Comment (nmadhok) [inline pillar.example]: I don't like this change. Plugin related pillar config should go under ``plugins`` pillar. It was correct earlier

Comment (SPoage) [inline pillar.example]: I don't disagree on your concept, but the pillar data in the example as it currently is in master is incorrect. I did not correct the underlying data model discrepancy - as that would be a backwards-incompatible modification - I simply corrected the documentation to reflect what actually is expected.

**Comment by iggy** [comment]
Outdated. Please rebase/resubmit.

**Pull request closed without merging**